nmigen/hdl/ast.py

```diff
@@ -417,6 +417,42 @@ def matches(self, *patterns):
         else:
             return Cat(*matches).any()
 
+    def rotate_left(self, offset):
+        """Rotate left by constant modulo 2**len(self).
+        
+        Parameters
+        ----------
+        offset : int, in
+            Amount to rotate by.
+
+        Returns
+        -------
+        Value, out
+            The input rotated left by offset if offset is positive, else the input rotated right by -offset.
+        """
+        if isinstance(offset, int):
```

Comment:
Could you invert this check for consistency with the rest of the code? If there are multiple precondition checks, the code is easier to understand that way; see `matches` for a good example of why.